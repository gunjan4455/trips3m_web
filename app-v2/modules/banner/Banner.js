import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { Helmet } from 'react-helmet';

import BannerContent from './BannerContent';
import Img from 'components/Common/Img';
import { createHelmetBanner } from 'helpers/seoHelpers';

const BannerContainer = styled.div(({ isLargeBanner }) => ({
  position: 'relative',
  height: isLargeBanner ? '270px' : '150px',
  overflow: 'hidden',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  '& img': {
    minWidth: 'auto',
    maxWidth: 'none',
    minHeight: '100%'
  },
  '&:after': {
    position: 'absolute',
    width: '100%',
    height: '100%',
    background: 'rgba(0,0,0,0.3)',
    top: '0',
    left: '0',
    content: '\' \'',
    zIndex: '1',
  }
}));

const Banner = ({ banner, isLargeBanner, canonicalUrl, isCityBlock, location }) => (
  <div className="clearfix">
    <Helmet>
      {createHelmetBanner(banner.metaTags, canonicalUrl, location.pathname)}
    </Helmet>
    <div className="clearfix sbc5 relative">
      <BannerContainer
        isLargeBanner={isLargeBanner}
        className="clearfix">
        {
          isCityBlock ?
            <Img
              width={500} height={270}
              src={banner.image} ikSrc={banner.imagekitUrl} alt={banner.altText} /> :
            <Img
              width={500} height={270}
              src={banner.image && banner.image.mobile}
              alt={banner.altText}
            />
        }
      </BannerContainer>
      <div className="absolute p15 l0 r0 b0 z2">
        <BannerContent banner={banner} />
      </div>
    </div>
  </div>
);

Banner.propTypes = {
  banner: PropTypes.object.isRequired,
  metaTags: PropTypes.array,
  location: PropTypes.object,
  showExploreForm: PropTypes.bool,
  showDuration: PropTypes.bool,
  bannerStyle: PropTypes.object,
  isLargeBanner: PropTypes.bool,
  canonicalUrl: PropTypes.string,
  isCityBlock: PropTypes.bool
};

Banner.defaultProps = {
  metaTags: [],
  showExploreForm: false,
  showDuration: false,
  bannerStyle: {},
  isLargeBanner: false,
  canonicalUrl: '',
  isCityBlock: false
};

export default Banner;
