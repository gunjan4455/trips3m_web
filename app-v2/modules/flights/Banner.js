import React from 'react';
import PropTypes from 'prop-types';

import PrefilledBanner from './PrefilledBanner';
import PostfilledBanner from './PostFilledBanner';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';


const Banner = ({ flightsData, trackSegment }) => {
  const trackSegmentEvent = (cta, object, miscType1) => {
    trackSegment({
      event: ORGANISM_CLICKED,
      section: 'Flights',
      cta,
      object,
      miscType1
    });
  };

  return(
    <div className="flex flexFull" style={{ maxWidth: '100%', minWidth: '100%' }}>
      {
        flightsData.length ?
          <PostfilledBanner trackSegmentEvent={trackSegmentEvent} />
          : <PrefilledBanner trackSegmentEvent={trackSegmentEvent} />
      }
    </div>
  );
};

Banner.propTypes = {
  flightsData: PropTypes.array.isRequired,
  trackSegment: PropTypes.func.isRequired
};

export default Banner;
