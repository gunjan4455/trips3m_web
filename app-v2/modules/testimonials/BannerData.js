import React from 'react';
import PropTypes from 'prop-types';

export const getReviewsPercentage = (positiveReview, totalReview) => (positiveReview / totalReview) * 100;

const BannerDataLeft = ({ positiveReview, totalReview, title }) => (
  <div className="absolute l15 b20 sfcw">
    <h1 className='m0 f32 sfcw'>{title}</h1>
    <p className="at_positivePercentage fw7 f24">
      <span className='f24 m0'>{getReviewsPercentage(positiveReview, totalReview).toFixed(2)}%{' '}</span>
      <span className="f16 fw4">Positive Reviews</span>
    </p>
    <p className='fw7 f16a'>Based On {totalReview} Ratings</p>
  </div>
);

BannerDataLeft.propTypes = {
  positiveReview: PropTypes.number.isRequired,
  totalReview: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

export default BannerDataLeft;
