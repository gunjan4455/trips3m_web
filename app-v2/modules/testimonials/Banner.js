import React from 'react';
import PropTypes from 'prop-types';

import BannerData from './BannerData';
import styled from '@emotion/styled';

const GImgDiv = styled.div({
  height: '270px',
  '& img': {
    minHeight: '100%',
    minWidth: '100%',
    maxWidth: 'none',
  }
});

const GOverlayDiv = styled.div({
  background: 'rgba(0,0,0,0.4)',
});

const Banner = ({ bannerPath, bannerAlt, totalRating, positiveRating }) => {
  return (
    <div className='relative'>
      <GImgDiv className='overflowh relative'>
        <img src={bannerPath} alt={bannerAlt} className='absolute-center' />
      </GImgDiv>
      <GOverlayDiv className='absolute b0 l0 r0 p15 t0'>
        <BannerData
          totalReview={totalRating}
          title={bannerAlt}
          positiveReview={positiveRating}
        />
      </GOverlayDiv>
    </div>
  );
};

Banner.propTypes = {
  bannerPath: PropTypes.string.isRequired,
  bannerAlt: PropTypes.string.isRequired,
  totalRating: PropTypes.number.isRequired,
  positiveRating: PropTypes.number.isRequired,
};

export default Banner;
