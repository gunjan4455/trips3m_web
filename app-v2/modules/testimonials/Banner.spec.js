import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Banner from './Banner';
import BannerData from './BannerData';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  bannerPath: '',
  bannerAlt: 'Happy Travelers',
  totalRating: 5430,
  positiveRating: 5172,
};

const shallowWrapperBanner = props => {
  return shallow(<Banner {...props} />);
};

describe('Testimonials module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperBanner(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass proper positive rating in bannerData', () => {
    const wrapper = shallowWrapperBanner(DEFAULT_PROPS);
    expect(wrapper.find(BannerData).prop('positiveReview')).toBe(DEFAULT_PROPS.positiveRating);
  });
});
