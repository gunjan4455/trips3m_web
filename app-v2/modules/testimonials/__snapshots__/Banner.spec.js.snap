// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Banner
    bannerAlt="Happy Travelers"
    bannerPath=""
    positiveRating={5172}
    totalRating={5430}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(render)
          className="overflowh relative"
        >
          <img
            alt="Happy Travelers"
            className="absolute-center"
            src=""
          />
        </ForwardRef(render)>,
        <ForwardRef(render)
          className="absolute b0 l0 r0 p15 t0"
        >
          <BannerDataLeft
            positiveReview={5172}
            title="Happy Travelers"
            totalReview={5430}
          />
        </ForwardRef(render)>,
      ],
      "className": "relative",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <img
            alt="Happy Travelers"
            className="absolute-center"
            src=""
          />,
          "className": "overflowh relative",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Happy Travelers",
            "className": "absolute-center",
            "src": "",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "& img": Object {
                "maxWidth": "none",
                "minHeight": "100%",
                "minWidth": "100%",
              },
              "height": "270px",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <BannerDataLeft
            positiveReview={5172}
            title="Happy Travelers"
            totalReview={5430}
          />,
          "className": "absolute b0 l0 r0 p15 t0",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "positiveReview": 5172,
            "title": "Happy Travelers",
            "totalReview": 5430,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "background": "rgba(0,0,0,0.4)",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(render)
            className="overflowh relative"
          >
            <img
              alt="Happy Travelers"
              className="absolute-center"
              src=""
            />
          </ForwardRef(render)>,
          <ForwardRef(render)
            className="absolute b0 l0 r0 p15 t0"
          >
            <BannerDataLeft
              positiveReview={5172}
              title="Happy Travelers"
              totalReview={5430}
            />
          </ForwardRef(render)>,
        ],
        "className": "relative",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <img
              alt="Happy Travelers"
              className="absolute-center"
              src=""
            />,
            "className": "overflowh relative",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Happy Travelers",
              "className": "absolute-center",
              "src": "",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              Object {
                "& img": Object {
                  "maxWidth": "none",
                  "minHeight": "100%",
                  "minWidth": "100%",
                },
                "height": "270px",
              },
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <BannerDataLeft
              positiveReview={5172}
              title="Happy Travelers"
              totalReview={5430}
            />,
            "className": "absolute b0 l0 r0 p15 t0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "positiveReview": 5172,
              "title": "Happy Travelers",
              "totalReview": 5430,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              Object {
                "background": "rgba(0,0,0,0.4)",
              },
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
